src/tuwien/auto/calimero/knxnetip/util/SRP.java

```diff
@@ -0,0 +1,220 @@
+/*
+    Calimero 2 - A library for KNX network access
+    Copyright (c) 2018 K.Heimrich
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program; if not, write to the Free Software
+    Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+
+    Linking this library statically or dynamically with other modules is
+    making a combined work based on this library. Thus, the terms and
+    conditions of the GNU General Public License cover the whole
+    combination.
+
+    As a special exception, the copyright holders of this library give you
+    permission to link this library with independent modules to produce an
+    executable, regardless of the license terms of these independent
+    modules, and to copy and distribute the resulting executable under terms
+    of your choice, provided that you also meet, for each linked independent
+    module, the terms and conditions of the license of that module. An
+    independent module is a module which is not derived from or based on
+    this library. If you modify this library, you may extend this exception
+    to your version of the library, but you are not obligated to do so. If
+    you do not wish to do so, delete this exception statement from your
+    version.
+*/
+
+package tuwien.auto.calimero.knxnetip.util;
+
+import tuwien.auto.calimero.KNXFormatException;
+import tuwien.auto.calimero.KNXIllegalArgumentException;
+
+/**
+ * Search Request Parameter Block (SRP).
+ * <p>
+ * A SRP is used to transfer additional information in a KNXnet/IP extended search requests.
+ * <p>
+ * @author Karsten Heimrich
+ */
+public final class SRP
```

Comment:
Got that.  :) Since I'm used to strong code style guide through my work, I was tempted to mimic as much of your code as possible. I've seen in newer commits that your style changed over time, so I will try to adapt to that.